src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java

```diff
@@ -178,7 +183,8 @@ public ResponseEntity updateDeckForAdmin(@RequestBody Deck deck, @PathVariable L
 
     @Auditable(action = AuditingAction.DELETE_DECK_ADMIN)
     @DeleteMapping(value = "/api/admin/decks/{deckId}")
-    public ResponseEntity deleteDeckForAdmin(@PathVariable Long deckId) {
+    public ResponseEntity deleteDeckForAdmin(@PathVariable Long deckId)throws NotAuthorisedUserException {
```

Comment:
Shouldn't this method be transactional? I am not sure but I think if the deletion of the deck generally wasn't successful, the deletion from the folder should be cancelled as well.

Reply:
ok

Reply:
Transactional should be on service level.